lib/ansible/modules/remote_management/irmc/irmc_facts.py

```diff
@@ -0,0 +1,520 @@
+             with relevant data (drives, firmware, id, name, volume)<br/>
+             <i>note that storage controllers are only returned if server is 'On'</i>
+        returned: always
+        type: dict
+        sample:
+            {
+                "count": 1,
+                "devices": [
+                    { "drives": 4, "firmware": "4.270.00-4869", "id": "1000", "name": "PRAID EP400i", "volumes": 1 }
+                ]
+            }
+    irmc:
+        description:
+             dict with relevant iRMC data
+             (fw_builddate, fw_running, fw_version, hostname, macaddress, sdrr_version)
+        returned: always
+        type: dict
+        sample:
+            {
+                "fw_builddate": "2018-03-05T14:02:44",
+                "fw_running": "LowFWImage",
+                "fw_version": "9.08F",
+                "hostname": "iRMC01CA5C-iRMC",
+                "macaddress": "90:1B:0E:01:CA:5C",
+                "sdrr_version": "3.73"
+            }
+    mainboard:
+        description:
+             dict with relevant mainboard data
+             (dnumber, manufacturer, part_number, serial_number, version)
+        returned: always
+        type: dict
+        sample:
+            {
+                "dnumber": "D3289",
+                "manufacturer": "FUJITSU",
+                "part_number": "S26361-D3289-D13",
+                "serial_number": "44617895",
+                "version": "WGS04 GS50"
+            }
+    system:
+        description:
+             dict with relevant system data
+             (asset_tag, bios_version, description, health, helpdesk_message, host_name,
+             idled_state, ip, location, manufacturer, memory_size, model, part_number,
+             power_state, serial_number, uuid)
+        returned: always
+        type: dict
+        sample:
+            {
+                "asset_tag": "New AssetTag",
+                "bios_version": "V5.0.0.9 R1.36.0 for D3289-A1x",
+                "contact": "Admin (admin@server.room)",
+                "description": "server description",
+                "health": "OK",
+                "helpdesk_message": "New helpdesk message",
+                "host_name": "STK-SLES11SP4x64",
+                "idled_state": "Off",
+                "ip": 101.102.103.104,
+                "location": "Server Room",
+                "manufacturer": "FUJITSU",
+                "memory_size": "24 GB",
+                "model": "PRIMERGY RX2540 M1",
+                "part_number": "ABN:K1495-VXXX-XX",
+                "power_state": "On",
+                "serial_number": "YLVT000098",
+                "uuid": "11223344-5566-cafe-babe-deadbeef1234"
+            }
+'''
+
+
+import json
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible.module_utils.irmc import irmc_redfish_get, irmc_redfish_patch, get_irmc_json
+
+
+def irmc_facts(module):
+    result = dict(
+        changed=False,
+        status=0
+    )
+
+    if module.check_mode:
```

Comment:
This code is not required as your module check mode is set to False.

Reply:
Can you make the module support checkmode?

Reply:
Will do.